packages/style-engine/README.md

```diff
@@ -258,6 +258,22 @@ _Changelog_
 
 `6.1.0` Introduced in WordPress core.
 
+### getCSSVarFromStyleValue
+
+Returns a WordPress CSS custom var value from incoming style preset value. The preset value follows the pattern `var:description|context|slug`.
+
+Example:
+
+`getCSSVarFromStyleValue( 'var:preset|color|heavenlyBlue' )` // returns 'var(--wp--preset--color--heavenly-blue)'
+
+_Parameters_
+
+-   _styleValue_ `string`: A raw style value.
+
+_Returns_
+
+-   `string`: string A CSS var value.
```

Comment:
This doesn't appear accurate.

Reply:
Given the test cases, I think we can refactor the typings. Instead of accepting only `string` `styleValue`, we seem to accept an unknown type.

```ts
function getCSSVarFromStyleValue<Type>( styleValue: Type ): Type
```

This seems to work in my testing (we might have to manual cast the return type in the function though).

If we only want to accept a list of types then we can narrow it and throw errors if needed.

Reply:
Yeah, so maybe we can use unions to show that the function _prefers_ to receive and return strings, but also acknowledge `any | unknown`, e.g., 

```ts
type GetCSSVarFromStyleValueFunction =
	| ( ( arg: string ) => string )
	| ( ( arg: any ) => unknown );
```